src/ExtraThemeAndPluginHeaders.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Class ExtraThemeAndPluginHeaders.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Registers the 'AMP' extra header for themes and plugins.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.0
+ * @internal
+ */
+final class ExtraThemeAndPluginHeaders implements Service, Registerable {
+
+	/**
+	 * Header name.
+	 *
+	 * @var string
+	 */
+	const AMP_HEADER = 'AMP';
+
+	/**
+	 * Values which indicate the theme has 'amp' theme support.
+	 *
+	 * @var string[]
+	 */
+	const AMP_THEME_SUPPORT_VALUES = [
+		'theme-support',
```

Comment:
Not a fan of multiple values having the same meaning for AMP compatibility; we might as well say if the header has a truthy value (except for `legacy`) then the theme supports AMP.

Also, since the use of header values might be used in the future to declare support for other AMP related features, I would recommend designing it in a way to allow for easy extension. For example, if in the future we would allow a theme to declare if it supports paired mode, I'd imagine a syntax such as:

```
AMP: mode=legacy,paired=false
```

Reply:
OK, I've opted for a simple truthy value in 5c16a78aa94a1dc966dd0d4ea7b9975cb37955fd